acquisitions/src/main/resources/thunderjet/mod-gobi/features/gobi-api-tests.feature

```diff
@@ -80,3 +80,214 @@ Feature: mod-gobi api tests
+    Then status 422
+
+    # Verify the newly added mapping
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method GET
+    Then status 200
+    And match response.orderMappings.orderType == 'UnlistedPrintSerial'
+    And match response.orderMappings.mappings[1].dataSource.default == "false"
+    And match response.orderMappings.mappings[2].dataSource.default == "true"
+
+    # Update existing custom mapping
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request valid_mapping
+    When method PUT
+    Then status 204
+
+    # Delete existing custom mapping
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method DELETE
+    Then status 200
+
+    # Verify original mapping
+    Given path '/gobi/orders/custom-mappings/UnlistedPrintSerial'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    When method GET
+    Then status 200
+    And match response.orderMappings.orderType == 'UnlistedPrintSerial'
+    And match response.orderMappings.mappings[1].dataSource.default == "true"
+    And match response.orderMappings.mappings[2].dataSource.default == "false"
+
+
+  Scenario: Make sure updated mapping is applied whenever placing orders
+    * def sample_po_original = read('classpath:samples/mod-gobi/po-unlisted-print-serial.xml')
+    * def sample_po_updated = read('classpath:samples/mod-gobi/po-unlisted-print-serial-updated.xml')
+
+    # Put an order for original mapping
+    Given path '/gobi/orders'
+    And headers { 'Content-Type': 'application/xml', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request sample_po_original
+    When method POST
+    Then status 201
+    * def poLineNumber = /Response/PoLineNumber
+
+    # Check order approved
+    Given path '/orders/composite-orders'
+    And headers headers
+    And param query = 'poNumber==*' + poLineNumber.split('-')[0]+'*'
+    When method GET
+    Then status 200
+    And match response.purchaseOrders[0].approved == true
+
+    # Update mapping
+    * def valid_mapping = read('classpath:samples/mod-gobi/unlisted-print-serial.json')
+    Given path '/gobi/orders/custom-mappings'
+    And headers { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request valid_mapping
+    When method POST
+    Then status 201
+
+    # Try to put an order for original mapping
+    Given path '/gobi/orders'
+    And headers { 'Content-Type': 'application/xml', 'x-okapi-token': '#(okapitoken)', 'Accept': '*/*' }
+    And request sample_po_original
+    When method POST
+    Then status 500
```

Comment:
if it is has some error code or message we can add assertion here.
It will also improve readability why test is failed in additional to testing error code provided